test/wallet.test.ts

```diff
@@ -0,0 +1,339 @@
+import { JsonRpcEngine } from 'json-rpc-engine';
+import pify from 'pify';
+
+import {
+  providerFromEngine,
+  createWalletMiddleware,
+  TransactionParams,
+  MessageParams,
+} from '../src';
+
+const testAddresses = [
+  '0xbe93f9bacbcffc8ee6663f2647917ed7a20a57bb',
+  '0x1234362ef32bcd26d3dd18ca749378213625ba0b',
+];
+const testUnkownAddress = '0xbadbadbadbadbadbadbadbadbadbadbadbadbad6';
+const testTxHash =
+  '0xceb3240213640d89419829f3e8011d015af7a7ab3b54c14fdf125620ce5b8697';
+const testMsgSig =
+  '0x68dc980608bceb5f99f691e62c32caccaee05317309015e9454eba1a14c3cd4505d1dd098b8339801239c9bcaac3c4df95569dcf307108b92f68711379be14d81c';
+
+function createTestSetup() {
+  const engine = new JsonRpcEngine();
+  const provider = providerFromEngine(engine);
+
+  return { engine, provider };
+}
+
+describe('wallet', () => {
+  describe('accounts', () => {
+    it('returns null for coinbase when no accounts', async () => {
+      const { engine } = createTestSetup();
+      const getAccounts = async () => [];
+      engine.push(createWalletMiddleware({ getAccounts }));
+      const coinbaseResult = await pify(engine.handle).call(engine, {
+        method: 'eth_coinbase',
+      });
+      expect(coinbaseResult.result).toEqual(null);
+    });
+
+    it('wallet coinbase returns the correct value from getAccounts', async () => {
+      const { engine } = createTestSetup();
+      const getAccounts = async () => testAddresses.slice();
+      engine.push(createWalletMiddleware({ getAccounts }));
+      const coinbaseResult = await pify(engine.handle).call(engine, {
+        method: 'eth_coinbase',
+      });
+      expect(coinbaseResult.result).toEqual(testAddresses[0]);
+    });
+
+    it('wallet coinbase returns the correct value from getAccounts with multiple accounts', async () => {
+      const { engine } = createTestSetup();
+      const getAccounts = async () => testAddresses.slice(0, 2);
+      engine.push(createWalletMiddleware({ getAccounts }));
+      const coinbaseResult = await pify(engine.handle).call(engine, {
+        method: 'eth_coinbase',
+      });
+      expect(coinbaseResult.result).toEqual(testAddresses[0]);
+    });
+  });
+
+  describe('transactions', () => {
+    it('valid address for processTransaction', async () => {
```

Comment:
What do you think about rewording this to describe the behavior and not merely how the test is being set up? I think there are some other examples in this file.